### PR TITLE
lib: nrf_cloud: Make connection_poll_thread stack size adjustible

### PR DESCRIPTION
**Pull request by glarsennordic**
Added `CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD_STACK_SIZE` KConfig option, which is now used to determine the stack size of `nrfcloud_connection_poll_thread`

IRIS-4765

Signed-off-by: Georges Oates_Larsen <georges.larsen@nordicsemi.no>

### CODE REVIEW

**Review by jayteemo — APPROVED** [review]

File: doc/nrf/libraries/networking/nrf_cloud.rst [hunk 1]
```diff
@@ -33,9 +33,12 @@ Connecting
...
 This API triggers a series of events and actions in the system.
 If the API fails, the application must retry to connect.
-If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket. :c:func:`nrf_cloud_connect` blocks and returns success when the MQTT connection to the cloud completes.
-If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket. :c:func:`nrf_cloud_connect` does not block and returns success if the connection monitoring thread has started.
+If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket.
+:c:func:`nrf_cloud_connect` blocks and returns success when the MQTT connection to the cloud completes.
+If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket.
+:c:func:`nrf_cloud_connect` does not block and returns success if the connection monitoring thread has started.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud.rst:39]:
```suggestion
If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket.
The :c:func:`nrf_cloud_connect` function blocks and returns success when the MQTT connection to the cloud completes.
If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket.
The :c:func:`nrf_cloud_connect` function does not block and returns success if the connection monitoring thread has started.
```

File: doc/nrf/libraries/networking/nrf_cloud.rst [hunk 2]
```diff
@@ -33,9 +33,12 @@ Connecting
...
 When :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` is enabled, an additional event, :c:enum:`NRF_CLOUD_EVT_TRANSPORT_CONNECTING`, is sent to the application.
+You can adjust the stack size of the connection monitoring thread by setting the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD_STACK_SIZE` Kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud.rst:41]:
```suggestion
To adjust the stack size of the connection monitoring thread, set the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD_STACK_SIZE` Kconfig option.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/networking/nrf_cloud.rst (hunk 2 above)
Comment (glarsennordic) [reply doc/nrf/libraries/networking/nrf_cloud.rst:41]: Ah, I was so close. I'll get the hang of it eventually

File: doc/nrf/libraries/networking/nrf_cloud.rst [hunk 3]
```diff
@@ -33,9 +33,12 @@ Connecting
 The application can use :c:func:`nrf_cloud_connect` to connect to the cloud.
 This API triggers a series of events and actions in the system.
 If the API fails, the application must retry to connect.
-If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket. :c:func:`nrf_cloud_connect` blocks and returns success when the MQTT connection to the cloud completes.
-If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket. :c:func:`nrf_cloud_connect` does not block and returns success if the connection monitoring thread has started.
+If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket.
+The :c:func:`nrf_cloud_connect` function blocks and returns success when the MQTT connection to the cloud completes.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud.rst:37]:
```suggestion

If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket.
The :c:func:`nrf_cloud_connect` function blocks and returns success when the MQTT connection to the cloud completes.

```
Suggesting to split this paragraph to smaller chunks for better readability.

File: doc/nrf/libraries/networking/nrf_cloud.rst [hunk 4]
```diff
@@ -33,9 +33,12 @@ Connecting
...
+If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket.
+The :c:func:`nrf_cloud_connect` function does not block and returns success if the connection monitoring thread has started.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud.rst:39]:
```suggestion
If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket.
The :c:func:`nrf_cloud_connect` function does not block and returns success if the connection monitoring thread has started.

```

File: doc/nrf/libraries/networking/nrf_cloud.rst [hunk 5]
```diff
@@ -33,9 +33,12 @@ Connecting
 The application can use :c:func:`nrf_cloud_connect` to connect to the cloud.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud.rst:33]:
```suggestion
The application can use the :c:func:`nrf_cloud_connect` function to connect to the cloud.
```

File: doc/nrf/libraries/networking/nrf_cloud.rst [hunk 6]
```diff
@@ -33,9 +33,12 @@ Connecting
...
-If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket. :c:func:`nrf_cloud_connect` does not block and returns success if the connection monitoring thread has started.
+If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is not enabled, the application should monitor the connection socket.
+The :c:func:`nrf_cloud_connect` function blocks and returns success when the MQTT connection to the cloud completes.
+If the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an nRF Cloud library thread monitors the connection socket.
+The :c:func:`nrf_cloud_connect` function does not block and returns success if the connection monitoring thread has started.
 When :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` is enabled, an additional event, :c:enum:`NRF_CLOUD_EVT_TRANSPORT_CONNECTING`, is sent to the application.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud.rst:40]:
```suggestion
When the :kconfig:option:`CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` Kconfig option is enabled, an additional event, :c:enum:`NRF_CLOUD_EVT_TRANSPORT_CONNECTING`, is sent to the application.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by peknis — APPROVED** [review]

File: doc/nrf/libraries/networking/nrf_cloud.rst (hunk 4 above)
Comment (peknis) [reply doc/nrf/libraries/networking/nrf_cloud.rst:40]: I'm afraid you have not implemented this split quite yet. I meant the second paragraph under the heading "Connecting" would be split into two paragraphs, ending up to two starting with "If the :kconfig:option:...

**Review by peknis — CHANGES_REQUESTED** [review]
Please do the split you seemed to like.

File: doc/nrf/libraries/networking/nrf_cloud.rst (hunk 4 above)
Comment (glarsennordic) [reply doc/nrf/libraries/networking/nrf_cloud.rst:40]: Sorry about that!

**Review by plskeggs — APPROVED** [review]

File: doc/nrf/libraries/networking/nrf_cloud.rst (hunk 4 above)
Comment (peknis) [reply doc/nrf/libraries/networking/nrf_cloud.rst:41]: No worries Georges.

**Review by peknis — APPROVED** [review]

**Pull request merged**